vyper/ast/pre_parser.py

```diff
@@ -43,6 +44,54 @@ def validate_version_pragma(version_str: str, start: ParserPosition) -> None:
             start,
         )
 
+class ForParserState(enum.Enum):
+    NOT_RUNNING = enum.auto()
+    START_SOON = enum.auto()
+    RUNNING = enum.auto()
+
+# a simple state machine which allows us to handle loop variable annotations
+# (which are rejected by the python parser due to pep-526, so we scoop up the
+# tokens between `:` and `in` and parse them and add them back in later).
+class ForParser:
+    def __init__(self):
+        self.annotations = {}
+        self._current_annotation = None
+
+        self._state = ForParserState.NOT_RUNNING
+        self._current_for_loop = None
+
+    def consume(self, token):
+        # state machine: we can start slurping tokens soon
+        if token.type == NAME and token.string == "for":
+            # note: self._is_running should be false here, but we don't sanity
+            # check here as that should be an error the parser will handle.
```

Comment:
`self._state` should be NOT_RUNNING?

Reply:
Ah yea- I had another version with different variable names